bftengine/include/bftengine/ControlStateManager.hpp

```diff
@@ -14,19 +14,53 @@
 #pragma once
 #include <optional>
 #include "IStateTransfer.hpp"
+#include "ReservedPages.hpp"
+#include "Serializable.h"
+#include "SysConsts.hpp"
 
 namespace bftEngine {
 
-class ControlStateManager {
+static constexpr uint32_t ControlHandlerStateManagerReservedPagesIndex = 1;
+static constexpr uint32_t ControlHandlerStateManagerNumOfReservedPages = 1;
+class ControlStateManager : public ResPagesClient<ControlStateManager,
+                                                  ControlHandlerStateManagerReservedPagesIndex,
+                                                  ControlHandlerStateManagerNumOfReservedPages> {
  public:
-  void setStopAtNextCheckpoint();
-  std::optional<uint64_t> getStopCheckpointToStopAt();
+  void setStopAtNextCheckpoint(int64_t currentSeqNum);
+  std::optional<int64_t> getCheckpointToStopAt();
 
-  ControlStateManager(IStateTransfer& state_transfer);
+  ControlStateManager(IStateTransfer* state_transfer, uint32_t sizeOfReservedPages);
   ControlStateManager& operator=(const ControlStateManager&) = delete;
   ControlStateManager(const ControlStateManager&) = delete;
+  ~ControlStateManager() {}
 
  private:
-  IStateTransfer& state_transfer_;
+  IStateTransfer* state_transfer_;
```

Comment:
What was the rationale for switching to a pointer? I never saw a response from @toly-kournik on this.

Reply:
All state transfer holders, are holding it via a raw pointer (for now).
@toly-kournik wants it to be the same in all.